### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by jasonkatonica**
All the project dependencies were updated to the latest versions.

Plugins not yet declared in the dependency section were added. This
allows for easier resolution of dependencies as a whole if the user
desires to pre-download all dependencies needed by the project.

Signed-off-by: Jason Katonica <katonica@us.ibm.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -458,24 +476,33 @@
...
-            <version>3.3.0</version>
+            <version>3.4.2</version>
             <type>maven-plugin</type>
             <exclusions>
                 <exclusion>
                     <groupId>org.codehaus.plexus</groupId>
                     <artifactId>plexus-archiver</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.iq80.snappy</groupId>
```
Comment (KostasTsiounis) [inline pom.xml:492]: I'm guessing this has a CVE and we need a more updated version, right?

File: pom.xml [hunk 2]
```diff
@@ -399,14 +392,39 @@
...
-                <version>5.10.2</version>
+                <version>5.10.3</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
         </dependencies>
     </dependencyManagement>
 
     <dependencies>
+        <dependency>
```
Comment (KostasTsiounis) [inline pom.xml:403]: How come we didn't need to explicitly add the dependencies for those in the past, but we do now?

File: pom.xml (hunk 1 above)
Comment (jasonkatonica) [reply pom.xml:492]: Yes correct.

File: pom.xml (hunk 2 above)
Comment (jasonkatonica) [reply pom.xml:403]: This is not necessarily a requirement. This allows us to declare dependencies up front such that users can pre-download all dependencies in our pom file. I am ok reverting these changes if we think there is a downside to doing this.

Comment (KostasTsiounis) [reply pom.xml:403]: Not an issue. Just wondering.

**Review by KostasTsiounis — APPROVED** [review]
LGTM

**Review by johnpeck-us-ibm — APPROVED** [review]

**Pull request merged**